patches/api/0446-Add-FluidState-API.patch

```diff
@@ -0,0 +1,201 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: vicisacat <victor.branchu@gmail.com>
+Date: Fri, 17 Nov 2023 20:21:47 +0100
+Subject: [PATCH] Add FluidState API
+
+
+diff --git a/src/main/java/io/papermc/paper/block/fluid/FluidData.java b/src/main/java/io/papermc/paper/block/fluid/FluidData.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..17c94569268c8998b159eba8ec160eea8d075a29
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/block/fluid/FluidData.java
+@@ -0,0 +1,63 @@
++package io.papermc.paper.block.fluid;
++
++import io.papermc.paper.math.Position;
++import org.bukkit.Fluid;
++import org.bukkit.Location;
++import org.bukkit.World;
++import org.bukkit.util.Vector;
++import org.jetbrains.annotations.NotNull;
++
++public interface FluidData extends Cloneable {
++
++    /**
++     * Gets the fluid type of this fluid data.
++     *
++     * @return the fluid type
++     */
++    @NotNull Fluid getFluidType();
++
++    /**
++     * Returns a copy of this FluidData.
++     *
++     * @return a copy of the fluid data
++     */
++    @NotNull FluidData clone();
++
++    /**
++     * Returns the flowing direction the liquid as a vector
++     *
++     * @return the flow direction vector
++     */
++    Vector getFlowDirection(Location location);
++
++    /**
++     * Returns the amount of liquid is in the fluid, as in the number of layers.
++     *
++     * @return the amount as an integer, between 0 and 8
++     */
++    int getAmount();
++
++    /**
++     * Returns the height of the fluid in blocks
++     *
++     * @return the height as a float value, representing the height in blocks
++     */
++    float getHeight();
++
++    /**
++     * Returns whether this fluid is the same fluid as the one passed in the parameter
++     *
++     * @param fluid a fluid
++     * @return true if they are the same, false otherwise
++     */
++    default boolean is(Fluid fluid) {
```

Comment:
I don't see the purpose of this event. We already have a getFluidType which does this check, not really needed and we don't have similar methods on other similar types.